eidas_node/proxy_service/views.py

```diff
@@ -304,12 +305,28 @@ def get_saml_response(
             response.in_response_to_id,
         )
 
-        if cert_file:
-            response.verify_response(cert_file)
+        if cert_files:
+            for cert_file in cert_files:
+                try:
+                    response.verify_response(cert_file)
+                except (SecurityError, XmlsecError):
+                    pass
+                else:
+                    break
+            else:
+                raise SecurityError("Wrong signature")
         if key_source and key_location:
             response.decrypt(key_source, key_location)
-        if cert_file:
-            response.verify_assertion(cert_file)
+        if cert_files:
+            for cert_file in cert_files:
+                try:
+                    response.verify_assertion(cert_file)
+                except (SecurityError, XmlsecError):
+                    pass
+                else:
+                    break
+            else:
+                raise SecurityError("Wrong assertion")
```

Comment:
This could be hard to debug as errors are lost, I would suggest to gather them in list and print them per each cert_file.